tests/components/renault/test_binary_sensor.py

```diff
@@ -0,0 +1,189 @@
+"""Tests for Renault binary sensors."""
+from unittest.mock import patch
+
+import pytest
+from renault_api.kamereon import exceptions
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    ATTR_SW_VERSION,
+    STATE_OFF,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.setup import async_setup_component
+
+from . import (
+    setup_renault_integration_vehicle,
+    setup_renault_integration_vehicle_with_no_data,
+    setup_renault_integration_vehicle_with_side_effect,
+)
+from .const import MOCK_VEHICLES
+
+from tests.common import mock_device_registry, mock_registry
+
+
+@pytest.mark.parametrize("vehicle_type", MOCK_VEHICLES.keys())
+async def test_binary_sensors(hass, vehicle_type):
+    """Test for Renault binary sensors."""
+    await async_setup_component(hass, "persistent_notification", {})
+    entity_registry = mock_registry(hass)
+    device_registry = mock_device_registry(hass)
+
+    with patch("homeassistant.components.renault.PLATFORMS", [BINARY_SENSOR_DOMAIN]):
+        await setup_renault_integration_vehicle(hass, vehicle_type)
+        await hass.async_block_till_done()
+
+    mock_vehicle = MOCK_VEHICLES[vehicle_type]
+    expected_entities = mock_vehicle[BINARY_SENSOR_DOMAIN]
+    if len(expected_entities) > 0:
+        assert len(device_registry.devices) == 1
+        expected_device = mock_vehicle["expected_device"]
+        registry_entry = device_registry.async_get_device(
+            expected_device[ATTR_IDENTIFIERS]
+        )
+        assert registry_entry is not None
+        assert registry_entry.identifiers == expected_device[ATTR_IDENTIFIERS]
+        assert registry_entry.manufacturer == expected_device[ATTR_MANUFACTURER]
+        assert registry_entry.name == expected_device[ATTR_NAME]
+        assert registry_entry.model == expected_device[ATTR_MODEL]
+        assert registry_entry.sw_version == expected_device[ATTR_SW_VERSION]
```

Comment:
We should not have an if here, it will hide issues

Reply:
Same goes for the other tests added here

Reply:
There are cases where the `expected_entities` is zero (ie. for fuel vehicles which do not have plug/charge statuses).
I have altered the logic so that if `expected_entities` is zero then we ensure the device and entity registries are both empty.